program-libs/compressed-account/src/indexer_event/parse.rs

```diff
@@ -286,15 +286,15 @@ fn deserialize_instruction<'a>(
         ));
     }
     let instruction_discriminator = instruction[0..8].try_into().unwrap();
-    let instruction = instruction.split_at(12).1;
+    let instruction = instruction.split_at(8).1;
     match instruction_discriminator {
         // Cannot be exucted with cpi context -> executing tx
         DISCRIMINATOR_INVOKE => {
             if accounts.len() < 9 {
                 return Err(ParseIndexerEventError::DeserializeSystemInstructionError);
             }
             let accounts = accounts.split_at(9).1;
-            let data = InstructionDataInvoke::deserialize(&mut &instruction[..])?;
+            let data = InstructionDataInvoke::deserialize(&mut &instruction[4..])?;
```

Comment:
i'd suggest we add a comment to clarify the magic number. something like
```suggestion
            // Skips vec size bytes for backward compatibility
            let data = InstructionDataInvoke::deserialize(&mut &instruction[4..])?;
```